utils/python/CIME/compare_namelists.py

```diff
@@ -71,11 +71,15 @@ def _interpret_value(value_str, filename):
         new_tokens = []
         for token in tokens:
             if "*" in token:
-                # the following ensure that the following to namelist settings trigger a match
-                # nmlvalue = 1,1,1 versus nmlvalue = 3*1
-                sub_tokens = [item.strip() for item in token.split("*")]
-                expect(len(sub_tokens) == 2, "Incorrect usage of multiplication in token '%s'" % token)
-                new_tokens.extend([sub_tokens[1]] * int(sub_tokens[0]))
+                try:
+                    # the following ensure that the following to namelist settings trigger a match
+                    # nmlvalue = 1,1,1 versus nmlvalue = 3*1
+                    sub_tokens = [item.strip() for item in token.split("*")]
+                    expect(len(sub_tokens) == 2, "Incorrect usage of multiplication in token '%s'" % token)
+                    new_tokens.extend([sub_tokens[1]] * int(sub_tokens[0]))
+                except:
+                    # User probably did not intend to use the * operator as a namelist multiplier
+                    new_tokens.append(token)
```

Comment:
That's what I figured but why the different processing for the two different expected "error" types?
Does the expect error message get printed out when inside a try? If not, why have it?
```
if len(sub_tokens) != 2:
  raise Error
```
gets the message across of what you are trying to accomplish. On the other hand, if you like the expect syntax, 
`expect(isinstance( <var>, ( int, long ) ), "Repeat count should be an integer")`
Finally, there is the null-value repeat count:
`6*,`
means six null values but your current code does not handle that type of situation.
BTW, do you have a test for null values (a null value is recognized as a comma with no preceding non-whitespace characters)?